### PR TITLE
feat(#52): add --wide-first startup option

### LINKED ISSUE #52: Wide layout Master is not Master
### Description

Hello, I want to admit that I use both river and wideriver for the first time. However I noticed a huge issue with Wide layout which makes this layout unusable for me.

I launch wideriver with those settings: `--layout wide --stack even --count-master 1 --ratio-wide 0.40 --count-wide-left 1`. 

Background:
When you use other layout, like `--layout left --stack even` or `--layout right --stack even` then Master window is always left or right and spawning new windows or launching `riverctl zoom` places window in Master area. For left Layout it goes to the left, and for right Layout is goes to the right and that is expected.

However, on wide layout, even when ratio works correctly (center window is wider than left and right windows), the Master area isn't on the center but on the left. New spawned windows all goes to the left Stack and launching `riverctl zoom` on e.g. window on the right stack, moves this window to the left Stack.

Expected behavior:
New spawned windows in wide Layout should step in for Master area (on the center!). `riverctl zoom` should also move selected window to the center area.

### river version

0.3.13

### wideriver version

1.3.1

### Logs

```text
I [10:49:12] wideriver version 1.3.1
I [10:49:12] --layout                       wide
I [10:49:12] --layout-alt                   monocle
I [10:49:12] --stack                        even
I [10:49:12] --count-master                 1
I [10:49:12] --ratio-master                 0.6
I [10:49:12] --count-wide-left              1
I [10:49:12] --ratio-wide                   0.4
I [10:49:12] --smart-gaps
I [10:49:12] --inner-gaps                   5
I [10:49:12] --outer-gaps                   10
I [10:49:12] --border-width                 2
I [10:49:12] --border-width-monocle         2
I [10:49:12] --border-width-smart-gaps      2
I [10:49:12] --border-color-focused         
I [10:49:12] --border-color-focused-monocle 
I [10:49:12] --border-color-unfocused       
I [10:49:12] --log-threshold                info
I [10:49:12] wideriver started
```

### Steps to reproduce

+ Launch wideriver with settings above
+ Spawn 3 or more windows
+ Observe that new windows are always spawned on the left Stack
+ Try to zoom any stack window to the Master area
+ Observe that window goes to the left Stack instead of the Master area

### PR DESCRIPTION
**Pull request by xdom**
Fixes #52

Arrange the views so that the first view in the list (the master window) is placed in the master area. Maintain the focus and swapping order by treating the end of the views list as the left stack.

Note: This introduces a minor breaking change. Previously, windows at the end of the stack were placed in the right area; now, they are placed in the left area. This may cause issues for anyone relying on scripts that open windows in a specific order and expect a particular window arrangement.

### CODE REVIEW

**Comment by alex-courtis** [comment]
Thanks for the PR, I'll get to a review soon.

**Comment by alex-courtis** [comment]
Looks like a cmocka update has broken tests. I'll get a fix on master shortly.

**Comment by alex-courtis** [comment]
I took the liberty of rebasing, CI is good now.

As this is a breaking change, we'll need to ensure that existing users are not affected by this.

Let's put this behind an option, with the old behaviour being the default, something like:

`--wide-first  mid|left   left`

File: tst/tst-layout.c
Comment (alex-courtis) [inline tst/tst-layout.c:1]: Many thanks for adding tests where there were none.

**Comment by xdom** [comment]
@alex-courtis I put the new behaviour behind the `--wide-first` "feature flag".

**Comment by alex-courtis** [comment]
Thanks for the update. I'll get to a review and test soon.

Don't worry about the include-what-you-use failure. I might just retire it as it breaks and changes behaviour every llvm update.

File: inc/enum.h [hunk 1]
```diff
@@ -28,6 +28,12 @@ enum LogThreshold {
 	LOG_THRESHOLD_DEFAULT = INFO,
 };
 
+enum WideFirst {
```
Comment (alex-courtis) [inline inc/enum.h:31]: Enum is good - leaves the door open for people to implement right/top/whatever.

**Review by alex-courtis — APPROVED** [review]
Many thanks for your contribution! Works exactly as advertised and well documented/tested.
